localstack-core/localstack/services/lambda_/invocation/logs.py

```diff
@@ -50,10 +50,34 @@ def run_log_loop(self, *args, **kwargs) -> None:
             log_item = self.log_queue.get()
             if log_item is QUEUE_SHUTDOWN:
                 return
+            # we need to split by newline - but keep the newlines in the strings
+            # strips empty lines, as they are not accepted by cloudwatch
+            logs = [line + "\n" for line in log_item.logs.split("\n") if line]
```

Comment:
Do we have a test for the empty lines case?

Reply:
No, but cloud watch does not accept empty lines. The logic matches the one in `store_cloudwatch_logs` though, so we at least do not get worse in terms of parity. We should write tests for this though.